CMakeLists.txt

```diff
@@ -173,7 +176,7 @@ if(BUILD_TESTS)
 endif()
 
 target_include_directories(azure-storage-lite PUBLIC ${PROJECT_SOURCE_DIR}/include ${CURL_INCLUDE_DIRS} PRIVATE ${EXTRA_INCLUDE_DIRS})
-target_link_libraries(azure-storage-lite Threads::Threads ${CURL_LIBRARIES} ${EXTRA_LIBRARIES})
+target_link_libraries(azure-storage-lite Threads::Threads ${CURL_LIBRARIES} ${EXTRA_LIBRARIES} dl)
```

Comment:
I think it may be better to use https://cmake.org/cmake/help/latest/variable/CMAKE_DL_LIBS.html

Reply:
Yes, use the ${CMAKE_DL_LIBS} variable.
e.g. `target_link_libraries(azure-storage-lite Threads::Threads ${CURL_LIBRARIES} ${EXTRA_LIBRARIES} ${CMAKE_DL_LIBS})`